### PR TITLE
feat: support modify sqlSource field index value

### PR DESCRIPTION
**Pull request by Yisaer**

### CODE REVIEW

**Review by ngjaying — COMMENTED** [review]
I prefer a general state read/write API, not specific to SQL index value.

File: extensions/sources/sql/ext/sql.go [hunk 1]
```diff
@@ -41,6 +42,9 @@ type sqlsource struct {
 	Query sqlgen.SqlQueryGenerator
 	// The db connection instance
 	db *sql.DB
+
+	RuleID     string
```
Comment (ngjaying) [inline extensions/sources/sql/ext/sql.go:46]: Actually, these information are already in the context. Is it possible not to add another ref? I don't think the SQL source buisiness logic relates to the rule name, stream name.

**Review by ngjaying — CHANGES_REQUESTED** [review]

File: extensions/sources/sql/ext/sql.go (hunk 1 above)
Comment (Yisaer) [reply extensions/sources/sql/ext/sql.go]: updated.

File: internal/server/rule_state.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	if s != rule.RuleStarted {
+		return fmt.Errorf("rule %v should be running when modify state", ruleID)
+	}
+
+	req := &resetOffsetRequest{}
+	if err := cast.MapToStruct(param, req); err != nil {
+		return err
+	}
+	switch strings.ToLower(req.OffsetType) {
+	case "sql":
```
Comment (ngjaying) [inline internal/server/rule_state.go:96]: Not sure if this is needed? Because the type is already controlled in Topo

**Review by ngjaying — APPROVED** [review]

**Pull request merged**